src/main.js

```diff
@@ -507,11 +507,12 @@ class MiniGraphCard extends LitElement {
   }
 
   renderSvg() {
-    const { height } = this.config;
+    const { height, grid_line_type = false } = this.config;
     return svg`
       <svg width='100%' height=${height !== 0 ? '100%' : 0} viewBox='0 0 500 ${height}'
         @click=${e => e.stopPropagation()}>
         <g>
+          ${grid_line_type ? this.renderGridLines() : ''}
```

Comment:
Probably next "layers" will "overlay" the grid & make it less visible. I am thinking about drawing the grid AFTER these layers. But let's leave it this way & test...

Reply:
I tried to move it around, but I don't see any visual changes to that

Reply:
> don't see any visual changes to that

OK, we'll test it together

Reply:
@JulienDeveaux 
@akloeckner 
Guys, let's discuss this.
Currently parts of a graph are shown in this order:
-- grid lines
-- graph curve (class `line--rect`) + fill (class `fill--rect`)
-- graph points (class `line--points`)
Check this - the grid lines are below a curve:
![image](https://github.com/user-attachments/assets/67ce3a82-b3c9-4e21-8c21-9205174bec07)
and it definitely below a fill which has some transparency - so we can see grid line under the fill.
So far it looks more or less great.

But if a user makes a fill opaque:
![image](https://github.com/user-attachments/assets/eb5cdd3c-7d3b-466b-94aa-32a1fc88771a)
with
```
        card_mod:
          style: |
            .fill {
              opacity: 1 !important;
            }
```
then the grid lines are not visible.

What we can do - we can place the grid lines right before the `<g.line--points>` element (it must capture mouse events).
So, the grid lines will be above the curve, above the fill - but below the points.

Not sure that this look will be great.
IMHO the best look will be this:
-- fill
-- grid lines
-- curve
-- points
But in this case we have to place all these `<line>` elements inside the same `<g>` element (like it is already done currently) - i.e. not in the separate `<g.grid-lines>` element how I suggested.
Or (just a guess) we can try splitting the "main" `<g>` element into separate parts for the curve & the fill.

What do you think?

Reply:
Technically, I would consider grid lines to be part of the plot's background, i.e. they would be drawn below everything else. Even opaque fills should still benefit from such lines, because we can still see them, where they "stick out" on top of the fill.

I just checked out a bit, how professional plot libraries like MATLAB, Python or pgfplots handle this. (Hoping to find some consensus standard to just follow.) And Python's matplotlib is the only candidate to put grid lines in the foreground. The others seem to put them in the background. So, no consensus, unfortunately. 🤷

Reply:
> where they "stick out" on top of the fill.

I think this is a proper way.
Currently our grid lines are below the fill.
So, to achieve this we will have to split the "main" `<g>` into "fill" & "line" parts - and then place the new `grid--lines` element between them.
This is a content of `<g>` for 1 entity:
![image](https://github.com/user-attachments/assets/04b9e925-06a2-4f08-9e1e-9f49cdf7427f)
I honestly have no idea what these `defs` are for.

Reply:
Just guessing: the Def's are the colour map. That might also be the reason why fill and line are in the same group (`<g>`).

Reply:
Do you mean `color_thresholds`? One `defs` for a curve, another one for a fill.

Reply:
Also, looking at the code, we could simply render grid lines into the main group in renderSvg. We could then choose the order easily. Or is there a need to place them before the group itself as proposed currently?

However, by "stick out" I meant: the grid is hidden behind the fill. But it sticks out between the header and the fill. So, we can still use grid lines for orientation, but they do not disturb the clean look of the graph too much.

Reply:
> render grid lines into the main group in renderSvg

Grid lines were inside the main `<g>`. I asked to place them into a separate `<g>`.

Reply:
> the grid is hidden behind the fill. But it sticks out between the header and the fill. So, we can still use grid lines for orientation, but they do not disturb the clean look of the graph too much.

I see, this could be a better way - from some point of view.
Let's keep it like it is already done:
-- grid lines in a separate `<g>`
-- this group is before the main `<g>`.
And we will may be improve it in future.

Reply:
> Do you mean `color_thresholds`? One `defs` for a curve, another one for a fill.

Yes. That's what I thought. But it's not really relevant to this PR if we stick to the order we have it right now.

> Let's keep it like it is already done:

Alright! :+1:

Reply:
@JulienDeveaux
Do you have something to add? Or we "resolve the conversation"?

Reply:
That's good for me